packages/adblocker/src/engine/engine.ts

```diff
@@ -834,19 +936,60 @@ export default class FilterEngine extends EventEmitter<
       isFilterExcluded: this.isFilterExcluded.bind(this),
     });
 
+    const context: CosmeticFilterMatchingContext = {
+      url,
+      domain,
+      hostname,
+
+      classes,
+      hrefs,
+      ids,
+
+      allowGenericHides,
+      allowSpecificHides,
+
+      getBaseRules,
+      getInjectionRules,
+      getExtendedRules,
+      getRulesFromDOM,
+      getRulesFromHostname,
+
+      isFilterExcluded: this.isFilterExcluded.bind(this),
+
+      matchType: MatchingContextType.CosmeticFilter,
+      reference,
+    };
+
+    for (const match of matches) {
+      this.emit('filter-matched', match, context);
+
+      // These are for backward-compatibility
+      if (match.isScriptInject()) {
+        this.emit('scriptlet-matched', match, context);
+      } else if (match.isCSS()) {
+        this.emit('style-rule-matched', match, context);
+      } else if (match.isExtended()) {
+        this.emit('extended-rule-matched', match, context);
+      }
+
+      if (exceptionMatches.has(match)) {
+        this.emit('filter-excluded', exceptionMatches.get(match)!, context);
+      }
+    }
+
     // Perform interpolation for injected scripts
     const scripts: string[] = [];
     for (const injection of injections) {
       const script = injection.getScript(this.resources.js);
       if (script !== undefined) {
-        this.emit('script-injected', script, url);
+        this.emit('script-injected', script, url, context);
```

Comment:
not sure if we need to change those old events, maybe it will be easier to only introduce the new ones?

Reply:
The argument is appended, therefore I assume there won't be a significant side-effect to old codebases. Actually, we still want to use old event names with contexts.

Reply:
Reverted. :+1: